pennylane/_qubit_device.py

```diff
@@ -1571,27 +1566,34 @@ def sample(self, observable, shot_range=None, bin_size=None, counts=False):
                 samples = sub_samples
 
         else:
-            # Replace the basis state in the computational basis with the correct eigenvalue.
-            # Extract only the columns of the basis samples required based on ``wires``.
-            samples = sub_samples[..., np.array(device_wires)]  # Add np.array here for Jax support.
-            powers_of_two = 2 ** np.arange(samples.shape[-1])[::-1]
-            indices = samples @ powers_of_two
-            indices = np.array(indices)  # Add np.array here for Jax support.
+            # get eigvals
             if isinstance(observable, MeasurementValue):
                 eigvals = self._asarray(
                     [observable[i] for i in range(2 ** len(observable.measurements))],
                     dtype=self.R_DTYPE,
                 )
-                samples = eigvals[indices]
             else:
                 try:
-                    samples = observable.eigvals()[indices]
+                    eigvals = observable.eigvals()
                 except qml.operation.EigvalsUndefinedError as e:
                     # if observable has no info on eigenvalues, we cannot return this measurement
                     raise qml.operation.EigvalsUndefinedError(
                         f"Cannot compute samples of {observable.name}."
                     ) from e
 
+            # special handling for observables with eigvals +1/-1 to enable JIT compatibility.
+            if np.array_equal(eigvals, [1.0, -1.0]):
```

Comment:
Just for a quick pedantic check --- we should never have to worry about floating-point errors bringing these above or below 1.0 and -1.0, right? In other words, there's no need to do allclose here instead?

Reply:
I don't think so - that would only happen if we "did something" with the numbers, right? Like, we might see it if `eigvals=2.0/2`, but not if they were just defined as `[1.0, -1.0]` and then not touched. Or is that incorrect?

Reply:
As long as we are defining then directly, and not pulling them out from some function-calls or eigensolver elsewhere, we should be good.

Reply:
cool, we are